plugins/woocommerce/client/admin/client/settings-payments/onboarding/providers/woopayments/steps/business-verification/components/embedded/index.tsx

```diff
@@ -0,0 +1,162 @@
+/**
+ * External dependencies
+ */
+import React, { useState, useEffect } from 'react';
+import {
+	loadConnectAndInitialize,
+	LoadError,
+	LoaderStart,
+	StripeConnectInstance,
+} from '@stripe/connect-js';
+import {
+	ConnectAccountOnboarding,
+	ConnectComponentsProvider,
+} from '@stripe/react-connect-js';
+
+/**
+ * Internal dependencies
+ */
+import { createKycAccountSession } from './hooks';
+import appearance from './appearance';
+import { OnboardingFields } from '../../types';
+import BannerNotice from '../../../../components/banner-notice';
+import { useOnboardingContext } from '../../../../data/onboarding-context';
+
+interface EmbeddedComponentProps {
+	onLoaderStart?: ( { elementTagName }: LoaderStart ) => void;
+	onLoadError?: ( { error, elementTagName }: LoadError ) => void;
+}
+
+interface EmbeddedAccountOnboardingProps extends EmbeddedComponentProps {
+	onboardingData: OnboardingFields;
+	onExit: () => void;
+	onStepChange?: ( step: string ) => void;
+	collectPayoutRequirements?: boolean;
+	isPoEligible?: boolean;
+}
+
+/**
+ * Hook to initialize Stripe Connect.
+ *
+ * @param onboardingData - Data required for onboarding.
+ * @param isPoEligible   - Whether the user is eligible for progressive onboarding.
+ *
+ * @return Returns stripeConnectInstance, error, and loading state.
+ */
+const useInitializeStripe = (
+	onboardingData: OnboardingFields,
+	isPoEligible: boolean
+) => {
+	const [ stripeConnectInstance, setStripeConnectInstance ] =
+		useState< StripeConnectInstance | null >( null );
+	const { currentStep } = useOnboardingContext();
+	const [ initializationError, setInitializationError ] = useState<
+		string | null
+	>( null );
+	const [ loading, setLoading ] = useState< boolean >( true );
+
+	useEffect( () => {
+		const initializeStripe = async () => {
+			try {
+				const accountSession = await createKycAccountSession(
+					onboardingData,
+					currentStep?.actions?.kyc_session?.href ?? '',
+					isPoEligible
+				);
+
+				// To-Do: Track the embedded component redirection event.
```

Comment:
Do we have an issue for this? If not, maybe it is worth creating one to make sure we handle it 🙂

Reply:
Yes, I have created an issue for it https://github.com/woocommerce/woocommerce/issues/56826.